src/routes/auth/sign.tsx

```diff
@@ -30,82 +27,35 @@ import {
 } from "@arconnect/components";
 import AnimatedQRScanner from "~components/hardware/AnimatedQRScanner";
 import AnimatedQRPlayer from "~components/hardware/AnimatedQRPlayer";
-import type Transaction from "arweave/web/lib/transaction";
 import Wrapper from "~components/auth/Wrapper";
 import Progress from "~components/Progress";
 import browser from "webextension-polyfill";
-import Head from "~components/popup/Head";
 import useSetting from "~settings/hook";
 import prettyBytes from "pretty-bytes";
 import Arweave from "arweave";
 import { defaultGateway } from "~gateways/gateway";
 import BigNumber from "bignumber.js";
+import { useCurrentAuthRequest } from "~utils/auth/auth.hooks";
+import { HeadAuth } from "~components/HeadAuth";
+import { AuthButtons } from "~components/auth/AuthButtons";
 
 export default function Sign() {
-  // sign params
-  const params = useAuthParams<{
-    url: string;
-    address: string;
-    transaction: StrippedTx;
-    collectionID: string;
-  }>();
+  const { authRequest, acceptRequest, rejectRequest } =
+    useCurrentAuthRequest("sign");
 
-  // reconstructed transaction
-  const [transaction, setTransaction] = useState<Transaction>();
-
-  useEffect(() => {
-    (async () => {
-      if (!params?.transaction) return;
-
-      // reset tx
-      setTransaction(undefined);
-
-      // request chunks
-      sendMessage("auth_listening", null, "background");
```

Comment:
This is now loaded from `auth.provider.ts` so that the transaction data is not lost when navigating to a different page.

Reply:
👍 
Already discussed in a call